pkg/arvo/sys/vane/clay.hoon

```diff
@@ -4303,6 +4521,35 @@
       [mos ..^$]
     ==
   ::
+  ?:  ?=([%back-index @ @ @ ~] tea)
+    ?+    +<.q.hin  ~|  %clay-backfill-index-strange  !!
+        %done
+      ?~  error.q.hin
+        [~ ..^$]
+      ::  TODO better error handling
+      ::
+      ~&  %clay-take-backfill-index-error^our^tea^tag.u.error.q.hin
+      %-  (slog tang.u.error.q.hin)
+      [~ ..^$]
+    ::
+        %lost
+      ~|  %clay-take-backfill-lost^our
+      ::  TODO better error handling
+      !!
+    ::
+        %boon
+      =+  ;;  =blob  payload.q.hin
+      ::
+      =/  her=ship   (slav %p i.t.tea)
+      =/  =desk      (slav %tas i.t.t.tea)
+      =/  index=@ud  (slav %ud i.t.t.t.tea)
+      ::
+      =^  mos  ruf
+        =/  den  ((de our now ski hen ruf) her desk)
+        abet:abet:(take-backfill:(foreign-update:den index) blob)
+      [mos ..^$]
+    ==
```

Comment:
This PR is not the time or the place... But I'd like to work through these error semantics with you sometime soon.